rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -788,6 +790,42 @@ protected void doCODESIZE() {
         program.step();
     }
 
+    protected void doEXTCODEHASH() {
+        if (computeGas) {
+            gasCost = GasCost.EXT_CODE_HASH;
+            spendOpCodeGas();
+        }
+
+        //EXECUTION PHASE
+        DataWord address = program.stackPop();
+
+        ActivationConfig.ForBlock activations = program.getActivations();
+        PrecompiledContracts.PrecompiledContract precompiledContract = precompiledContracts.getContractForAddress(activations, address);
+        boolean isPrecompiledContract = precompiledContract != null;
+
+        if (isPrecompiledContract) {
+            program.stackPush(DataWord.valueOf(Keccak256Helper.keccak256(EMPTY_BYTE_ARRAY)));
+        }
+
+        Keccak256 codeHash = program.getCodeHashAt(address);
+
+        if (!isPrecompiledContract) {
+            //If account does not exist, 0 is pushed in stack
+            if (codeHash.equals(Keccak256.ZERO_HASH)) {
+                program.stackPush(DataWord.ZERO);
+            } else {
+                DataWord word = DataWord.valueOf(codeHash.getBytes());
+                program.stackPush(word);
+            }
+        }
+
+        if (isLogEnabled) {
+            hint = "hash: " + codeHash;
```

Comment:
Is it possible that, in the case of a precompiled contract, the `codeHash` value stored in the _hint_ is wrong?

Reply:
Yes, that was a problem, it was solved [here](https://github.com/rsksmart/rskj/blob/6833151421e39bcc84b4a37ef594e32a75b02699/rskj-core/src/main/java/org/ethereum/vm/VM.java#L810)